PPU - fix runtime overflow in get_sprite_pixel

Optimoos opened the pull request:
Per the fixme, there's got to be a better way to handle this that I just don't know about.

Optimoos committed:
PPU - fix runtime overflow in get_sprite_pixel

Per the fixme, there's got to be a better way to handle this that I just don't know about.

sophiajt commented on src/ppu.jakt:
```diff
@@ -12,7 +12,16 @@ class Sprite {
-        guard x >= .sprite_x and x < (.sprite_x + 8) else {
+
+
+        //FIXME: surely there is a more elegant way to handle overflows
```
we could make spritemax and sprite_x u16 instead of u8

Optimoos committed:
PPU - use larger datatypes to handle overflows

Still not certain this is the best way to do things. At the very least it
doesn't mirror what the hardware would do, but for now it stops some crashes.

jntrnr commented:
I mean make the variables u_16. Or does that cause us to add too many casts?

Optimoos commented:
I added spritemax just to deal with the overflow logic, it's not necessary if we use u16. sprite_x is expected to be u8 elsewhere so this was the compromise I came up with.
